JavaSource/org/unitime/timetable/gwt/client/departments/DepartmentsPage.java

```diff
@@ -0,0 +1,264 @@
+				// load
+				LoadingWidget.getInstance().show(MESSAGES.waitLoadingData());
+
+				// Get all departments
+				listDepartments();
+			}
+		});
+
+		// load
+		LoadingWidget.getInstance().show(MESSAGES.waitLoadingData());
+
+		// Get all departments
+		listDepartments();
+
+		// UI to update, add or back
+		iDepartmentsEdit = new DepartmentsEdit() {
+			
+			//Override on back event from DepartmentEdit page
+			@Override
+			protected void onBack(boolean refresh, final Long DepartmentId) {
+				iPanel.setWidget(iListDepartmentsForm);
+				UniTimePageLabel.getInstance().setPageName(
+						MESSAGES.pageDepartments());
+				if (refresh) {
+					LoadingWidget.getInstance().show(MESSAGES.waitLoadingData());
+					RPC.execute(new GetDepartmentsRequest(),new AsyncCallback<DepartmentsDataResponse>() {
+								@Override
+								public void onSuccess(DepartmentsDataResponse result) {
+									iListDepartmentsHeader.setEnabled("add",result.isCanAdd());
+									iListDepartmentsHeader.setEnabled("export",result.isCanExportPdf());
+									iListDepartmentsHeader.setEnabled("updateData", result.isCanUpdate());
+									iDepartmentsTable.setData(result.getDepartments(),iShowAllDept.getWidget().getValue());
+									LoadingWidget.getInstance().hide();
+									if (DepartmentId != null)
+										for (int i = 0; i < iDepartmentsTable.getRowCount(); i++) {
+											DepartmentInterface b = iDepartmentsTable.getData(i);
+											if (b != null && b.getId().equals(DepartmentId)) {
+												iDepartmentsTable.getRowFormatter().getElement(i).scrollIntoView();
+												iDepartmentsTable.setSelected(i, true);
+												break;
+											}
+										}
+								}
+
+								@Override
+								public void onFailure(Throwable caught) {
+									LoadingWidget.getInstance().hide();
+									UniTimeNotifications.error(MESSAGES.failedLoadData(caught.getMessage()), caught);
+									iListDepartmentsHeader.setErrorMessage(MESSAGES.failedLoadData(caught.getMessage()));
+									ToolBox.checkAccess(caught);
+								}
+
+							});
+				} else {
+					if (iDepartmentsTable.getSelectedRow() >= 0)
+						iDepartmentsTable.setSelected(iDepartmentsTable.getSelectedRow(), false);
+					if (DepartmentId != null)
+						for (int i = 0; i < iDepartmentsTable.getRowCount(); i++) {
+							DepartmentInterface b = iDepartmentsTable.getData(i);
+							if (b != null && b.getId().equals(DepartmentId)) {
+								iDepartmentsTable.getRowFormatter().getElement(i).scrollIntoView();
+								iDepartmentsTable.setSelected(i, true);
+								break;
+							}
+						}
+				}
+			}
+
+			//Override isAbbreviationUnique from DepartmentEdit page
+			@Override
+			protected boolean isAbbreviationUnique(
+					DepartmentInterface Department) {
+				for (int i = 0; i < iDepartmentsTable.getRowCount(); i++) {
+					DepartmentInterface b = iDepartmentsTable.getData(i);
+					if (b != null && !b.getId().equals(Department.getId()) && b.getAbbreviation().equalsIgnoreCase(
```

Comment:
While the user interface does not allow for null department abbreviation, the database does. So, the code should check for d.getAbbreviation() != null to avoid NullPointerException.